lib/processors/minifier.js

```diff
@@ -38,6 +42,76 @@ async function minifyInWorker(options, taskUtil) {
 	return getPool(taskUtil).exec("execMinification", [options]);
 }
 
+async function getSourceMap(resource, readFile) {
+	const resourceContent = await resource.getString();
+	const resourcePath = resource.getPath();
+
+	// This code is almost identical to lbt/bundle/Builder.js
+	// Please try to update both places when making improvements or bug fixes
+
+	const sourceMapUrlMatch = resourceContent.match(sourceMappingUrlPattern);
+	if (sourceMapUrlMatch) {
+		const sourceMapUrl = sourceMapUrlMatch[1];
+		log.silly(`Found source map reference in content of resource ${resourcePath}: ${sourceMapUrl}`);
+
+		// Strip sourceMappingURL from resource code to be minified
+		// It has no effect and would be wrong if addSourceMappingUrl == true
+		// In addition, as the debug-variant is created afterwards it does not have to take care to replace the comment
+		resource.setString(resourceContent.replace(sourceMappingUrlPattern, ""));
+
+		if (sourceMapUrl) {
+			if (sourceMapUrl.startsWith("data:")) {
+				// Data-URI indicates an inline source map
+				const expectedTypeAndEncoding = "data:application/json;charset=utf-8;base64,";
+				if (sourceMapUrl.startsWith(expectedTypeAndEncoding)) {
+					const base64Content = sourceMapUrl.slice(expectedTypeAndEncoding.length);
+					// Create a resource with a path suggesting it's the source map for the resource
+					// (which it is but inlined)
+					return Buffer.from(base64Content, "base64").toString();
+				} else {
+					log.warn(
+						`Source map reference in resource ${resourcePath} is a data URI but has an unexpected` +
+						`encoding: ${sourceMapUrl}. Expected it to start with ` +
+						`"data:application/json;charset=utf-8;base64,"`);
+				}
+			} else if (httpPattern.test(sourceMapUrl)) {
+				log.warn(`Source map reference in resource ${resourcePath} is an absolute URL. ` +
+					`Currently, only relative URLs are supported.`);
+			} else if (posixPath.isAbsolute(sourceMapUrl)) {
+				log.warn(`Source map reference in resource ${resourcePath} is an absolute path. ` +
+					`Currently, only relative paths are supported.`);
+			} else {
+				const sourceMapPath = posixPath.join(posixPath.dirname(resourcePath), sourceMapUrl);
+
+				try {
+					const sourceMapContent = await readFile(sourceMapPath);
+					if (sourceMapContent) {
+						return sourceMapContent.toString();
+					} else {
+						throw new Error(`Not found: ${sourceMapPath}`);
+					}
+				} catch (e) {
+					// No input source map
+					log.warn(`Unable to read source map for resource ${resourcePath}: ${e.message}`);
+				}
+			}
+		} else {
+			const sourceMapFileCandidate = resourcePath + ".map";
```

Comment:
We decided to remove this path, as it's not expected to have a Source Map file without a reference in the JavaScript file. Furthermore, if this is the case, it might be on purpose so that the existing Source Map is not used as input Source Map.

Reply:
For the records: according to the source map specs, a server can send an HTTP header to provide a reference to the source map for a resource it delivers. This is the only case where a source map would be needed without a reference in the code. 

In the context of UI5, however, we're not aware of any scenario that makes use of this feature and therefore currently don't consider it as a use case for the tooling.